include/z64scene.h

```diff
@@ -213,58 +216,132 @@ typedef struct {
     /* 0x4 */ void* segment;
 } SCmdMinimapChests; // size = 0x8
 
-typedef struct {
-    /* 0x0 */ Gfx* opaqueDl;
-    /* 0x4 */ Gfx* translucentDl;
-} RoomMeshType0Params; // size = 0x8
+typedef enum {
+    /* 0 */ ROOM_SHAPE_TYPE_NORMAL,
+    /* 1 */ ROOM_SHAPE_TYPE_IMAGE,
+    /* 2 */ ROOM_SHAPE_TYPE_CULLABLE,
+    /* 3 */ ROOM_SHAPE_TYPE_NONE,
+    /* 4 */ ROOM_SHAPE_TYPE_MAX
+} RoomShapeType;
 
-// Fields TODO
 typedef struct {
     /* 0x0 */ u8 type;
-    /* 0x1 */ u8 format; // 1 = single, 2 = multi
-} RoomMeshType1; // size = 0x2
-
-// Size TODO
-typedef struct {
-    /* 0x0 */ UNK_TYPE1 pad0[0x10];
-} RoomMeshType1Params; // size = 0x10
+} RoomShapeBase; // size = 0x1
 
 typedef struct {
-    /* 0x0 */ UNK_TYPE1 pad0[0x10];
-} RoomMeshType2Params; // size = 0x10
+    /* 0x0 */ Gfx* opa;
+    /* 0x4 */ Gfx* xlu;
+} RoomShapeDListsEntry; // size = 0x8
 
 typedef struct {
-    /* 0x0 */ u8 type;
-    /* 0x1 */ u8 count;
-    /* 0x2 */ UNK_TYPE1 pad2[0x2];
-    /* 0x4 */ RoomMeshType0Params* paramsStart;
-    /* 0x8 */ RoomMeshType0Params* paramsEnd;
-} RoomMeshType0; // size = 0xC
+    /* 0x0 */ RoomShapeBase base;
+    /* 0x1 */ u8 numEntries;
+    /* 0x4 */ RoomShapeDListsEntry* entries;
+    /* 0x8 */ RoomShapeDListsEntry* entriesEnd;
+} RoomShapeNormal; // size = 0xC
 
-typedef struct {
-    /* 0x0 */ u8 type;
-    /* 0x1 */ u8 count;
-    /* 0x2 */ UNK_TYPE1 pad2[0x2];
-    /* 0x4 */ RoomMeshType2Params* paramsStart;
-    /* 0x8 */ RoomMeshType2Params* paramsEnd;
-} RoomMeshType2; // size = 0xC
+typedef enum {
+    /* 1 */ ROOM_SHAPE_IMAGE_AMOUNT_SINGLE = 1,
+    /* 2 */ ROOM_SHAPE_IMAGE_AMOUNT_MULTI
+} RoomShapeImageAmountType;
+
+typedef struct {
+    /* 0x0 */ RoomShapeBase base;
+    /* 0x1 */ u8 amountType; // RoomShapeImageAmountType
+    /* 0x4 */ RoomShapeDListsEntry* entry;
+} RoomShapeImageBase; // size = 0x8
+
+typedef struct {
+    /* 0x00 */ RoomShapeImageBase base;
+    /* 0x08 */ void* source;
+    /* 0x0C */ u32 unk_0C;
+    /* 0x10 */ void* tlut;
+    /* 0x14 */ u16 width;
+    /* 0x16 */ u16 height;
+    /* 0x18 */ u8 fmt;
+    /* 0x19 */ u8 siz;
```

Comment:
Presumably there's a good reason these are abbreviations... named after gbi stuff?

Reply:
This was taken from OoT, and given `RoomShapeImageSingle` goes entirely unused in MM, this in a sense is dependent on OoT for docs. So that'll be a @Dragorn421 question since they did these docs for OoT. But I'm also assuming it's named after gbi stuff

Reply:
these abbreviations are how they are commonly named in gbi macro args yes. for example
`#define gDPLoadTextureBlock(pkt, timg, fmt, siz, width, height, pal, cms, cmt, masks, maskt, shifts, shiftt)`